tools/e2etesting/DeploySimulation.ps1

```diff
@@ -1,116 +1,160 @@
-Write-Host "Created edge device identity $($edgeIdentity.deviceId)"
-
-# Store Device Id within KeyVault to be useable by rerun without deployment (USeExisting=true)
-$secretvalue = ConvertTo-SecureString $edgeIdentity.deviceId -AsPlainText -Force
-$secret = Set-AzKeyVaultSecret -VaultName $keyVaultName -Name 'iot-edge-device-id' -SecretValue $secretvalue
-if (-Not $secret.Enabled) {
-    Write-Error "Couldn't store device identity in KeyVault" -ErrorAction Stop
+$iotHub = Get-AzIotHub -ResourceGroupName $ResourceGroupName
+
+if ($iotHub.Count -ne 1) {
+    Write-Error "IotHub could not be automatically selected in Resource Group '$($ResourceGroupName)'."    
+    return
+}
+
+Write-Host "IoT Hub Name: $($iotHub.Name)"
+
+$keyVault = Get-AzKeyVault -ResourceGroupName $ResourceGroupName
+
+if ($keyVault.Count -ne 1) {
+    Write-Error "keyVault could not be automatically selected in Resource Group '$($ResourceGroupName)'."    
+    return
+} 
+
+Write-Host "Key Vault Name: $($keyVault.VaultName)"
+
+$deviceName = "e2etestdevice_$($testSuffix)"
+
+Write-Host "Iot Hub Device Identity: $($deviceName)"
+
+$edgeIdentity = Get-AzIotHubDevice -ResourceGroupName $ResourceGroupName -IotHubName $iotHub.Name -DeviceId $deviceName -ErrorAction SilentlyContinue
+
+if (!$edgeIdentity) {
+    Write-Host "Creating edge-enabled device identity $($deviceName) in Iot Hub $($iotHub.Name)"
+    $edgeIdentity = Add-AzIotHubDevice -ResourceGroupName $ResourceGroupName -IotHubName $iotHub.Name -DeviceId $deviceName -EdgeEnabled
+}
+
+if (!$edgeIdentity.Capabilities.IotEdge) {
+    Write-Error "Device '$($edgeIdentity.Id)' Iot Hub: '$($iotHub.Name)') is not edge-enabled."
+    return
 }
 
-# Read Device Connection String for TestDevice
-$edgeDeviceConnectionString = az iot hub device-identity connection-string show --device-id $edgeIdentity.deviceId --hub-name $iotHub.Name | ConvertFrom-Json
-
-# Update Device Twin of IoT Edge device
-$deviceTwin = az iot hub device-twin show --device-id $edgeIdentity.deviceId --hub-name $iotHub.Name | ConvertFrom-Json
-$tags = "" | Select os,__type__
-$tags.os = "Linux"
-$tags.__type__ = "iiotedge"
-$deviceTwin | Add-Member -MemberType NoteProperty -Value $tags -Name "tags"
-$configFileName = New-TemporaryFile
-($deviceTwin | ConvertTo-Json -Depth 10) | Out-File -FilePath $configFileName -Encoding UTF8
-$out = az iot hub device-twin replace --device-id $edgeIdentity.deviceId --hub-name $iotHub.Name --json $configFileName
-Write-Host "Added Industrial IoT tags to Device Twin of edge device"
-
-$vmpassword = -join ((65..90) + (97..122) + (33..38) + (40..47) + (48..57)| Get-Random -Count 20 | % {[char]$_})
-$numberOfSimulations = 18
+Write-Host "Adding/Updating KeyVault-Secret 'iot-edge-device-id' with value '$($edgeIdentity.Id)'..."
+Set-AzKeyVaultSecret -VaultName $keyVault.VaultName -Name 'iot-edge-device-id' -SecretValue (ConvertTo-SecureString $edgeIdentity.Id -AsPlainText -Force) | Out-Null
```

Comment:
Why not check, if this was successfully?

Reply:
-> global Stop handler.